app/cli/internal/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,145 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry_test
+
+import (
+	"context"
+	"runtime"
+	"testing"
+
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestTags_WithEnvironmentInfo(t *testing.T) {
+	tags := telemetry.Tags{}.WithRuntimeInfo()
+
+	assert.Equal(t, runtime.GOOS, tags["os"])
+	assert.Equal(t, runtime.GOARCH, tags["arch"])
+}
+
+func TestCommandTracker_Track(t *testing.T) {
+	mockedClient := mocks.NewClient(t)
+	expectedEventName := "command_executed" // nolint: goconst
+
+	mockedClient.
+		On("TrackEvent", mock.Anything, "command_executed", mock.Anything, mock.Anything).
+		Return(func(_ context.Context, eventName string, id string, tags telemetry.Tags) error {
+			assert.NotEmpty(t, id)
+			assert.Equal(t, expectedEventName, eventName)
+
+			assert.Contains(t, tags, "os")
+			assert.Contains(t, tags, "arch")
+			assert.Contains(t, tags, "ci")
+			assert.NotContains(t, tags, "runner")
+			assert.Contains(t, tags, "command")
+
+			return nil
+		})
+	tracker := telemetry.NewCommandTracker(mockedClient)
+	assert.NotNil(t, tracker)
+
+	err := tracker.Track(context.Background(), "test-command", telemetry.Tags{})
+	assert.NoError(t, err)
+
+	mockedClient.AssertNumberOfCalls(t, "TrackEvent", 1)
+	mockedClient.AssertCalled(t, "TrackEvent", mock.Anything, expectedEventName, mock.Anything, mock.Anything)
+}
+
+func TestCommandTracker_Track_WithTags(t *testing.T) {
+	mockedClient := mocks.NewClient(t)
+	expectedEventName := "command_executed"
+
+	mockedClient.
+		On("TrackEvent", mock.Anything, "command_executed", mock.Anything, mock.Anything).
+		Return(func(_ context.Context, eventName string, id string, tags telemetry.Tags) error {
+			assert.NotEmpty(t, id)
+			assert.Equal(t, expectedEventName, eventName)
+
+			assert.Contains(t, tags, "os")
+			assert.Contains(t, tags, "arch")
+			assert.Contains(t, tags, "ci")
+			assert.NotContains(t, tags, "runner")
+			assert.Contains(t, tags, "command")
+			assert.Contains(t, tags, "tag1")
+			assert.Contains(t, tags, "tag2")
+
+			return nil
+		})
+	tracker := telemetry.NewCommandTracker(mockedClient)
+	assert.NotNil(t, tracker)
+
+	err := tracker.Track(context.Background(), "test-command", telemetry.Tags{
+		"tag1": "value1",
+		"tag2": "value2",
+	})
+	assert.NoError(t, err)
+
+	mockedClient.AssertNumberOfCalls(t, "TrackEvent", 1)
+	mockedClient.AssertCalled(t, "TrackEvent", mock.Anything, expectedEventName, mock.Anything, telemetry.Tags{
+		"os":      runtime.GOOS,
+		"arch":    runtime.GOARCH,
+		"ci":      "false",
+		"command": "test-command",
+		"tag1":    "value1",
+		"tag2":    "value2",
+	})
+}
+
+func TestCommandTracker_TrackWithEnvironmentInfo(t *testing.T) {
```

Comment:
I think we could simplify these tests. Basically call track without or with custom tags. That should be enough.

Then, you can have a unit test that properly tests the runner labels generation, which doesn't require to mock anything.

Does it make sense?

Reply:
Sure thing, let me add them.